indexing-service/src/main/java/org/apache/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -177,25 +179,13 @@ public Response taskPost(final Task task, @Context final HttpServletRequest req)
 
     return asLeaderWith(
         taskMaster.getTaskQueue(),
-        new Function<TaskQueue, Response>()
-        {
-          @Override
-          public Response apply(TaskQueue taskQueue)
-          {
-            try {
-              taskQueue.add(task);
-              return Response.ok(ImmutableMap.of("task", task.getId())).build();
-            }
-            catch (EntryExistsException e) {
-              return Response.status(Response.Status.BAD_REQUEST)
-                             .entity(
-                                 ImmutableMap.of(
-                                     "error",
-                                     StringUtils.format("Task[%s] already exists!", task.getId())
-                                 )
-                             )
-                             .build();
-            }
+        taskQueue -> {
+          try {
+            taskQueue.add(task);
+            return Response.ok(ImmutableMap.of("task", task.getId())).build();
+          }
+          catch (EntryExistsException e) {
+            throw new BadRequestException(StringUtils.format("Task[%s] already exists!", task.getId()));
```

Comment:
Earlier, this was returning a response like `return Response.status(BAD_REQUEST).entity().build()`. Shouldn't we continue to return a `Response` here instead of throwing an exception?